### PR TITLE
reflector: extract watch and startResyncAsync methods

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This is a no-op PR. It paves the way for #110772.

It extract logic from the `ListAndWatch` to separate methods.

We already have `list` and `watchHandler`. This PR add `watch` and `startResyncAsync`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/reflector.go [hunk 1]
```diff
@@ -323,6 +323,12 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	resyncerrc := make(chan error, 1)
 	cancelCh := make(chan struct{})
 	defer close(cancelCh)
+	r.startResyncAsync(stopCh, cancelCh, resyncerrc)
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/cache/reflector.go:326]: Actually, maybe let's switch to to:
```
go r.startResync(stopCh, cancelCh, resyncerrc)
```

and switch the startResync to get rid of `go() func() {}`

Comment (p0lyn0mial) [reply staging/src/k8s.io/client-go/tools/cache/reflector.go]: done

**Comment by wojtek-t** [comment]
/lgtm

Thanks!

**Pull request merged**